src/lib/format-time.js

```diff
@@ -0,0 +1,114 @@
+/**
+ Given a timestamp in the future, calculate the largest, closest unit to show.
+ On the high end we stop at days. e.g. 56 days is still counted in days not months or weeks.
+ On the low end we stop at minutes.
+ @param {number} timeStamp A future time stamp in ms.
+ @returns {object} containing the unit (min, hours, days) and how many. e.g. {unit: minutes, duration: 3}
+ */
+const getTopLevelTimeUnit = timeStamp => {
+    const diff = timeStamp - Date.now();
+    const oneDayInMs = 1000 * 60 * 60 * 24;
+    const oneHourInMs = 1000 * 60 * 60;
+    const oneMinuteInMs = 1000 * 60;
+    let unit = 'minutes';
+    let duration = diff / oneMinuteInMs;
+    if (diff > oneDayInMs) {
+        unit = 'days';
+        duration = diff / oneDayInMs;
+    } else if (diff > oneHourInMs) {
+        unit = 'hours';
+        duration = diff / oneHourInMs;
+    }
+    return {
+        unit: unit,
+        duration: duration
+    };
+};
+
+/**
+* This is used along with Intl.RelativeTimeFormat's formatToParts to extract
+* the value from a given key.
+* See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/formatToParts
+* Also see examples in format-time.test.jsx.
+* @param {string} key The name of the key you want to look up
+* @param {Array} parts An array of parts from formatToParts.
+* @returns {string} The value associated with the given key.
+*/
+const getValueFromKey = (key, parts) => {
+    for (const item of parts) {
+        if (item.type === key) {
+            return item.value;
+        }
+    }
+    // Just in case key doesn't exist.
+    return '';
+};
+
+/**
+* Given a future timestamp and a langauge, constructs a phrase to describe that time relative to now.
+* e.g. 2 days 3 hours, 3 minutes, 7 hours 38 minutes, 2 horas 5 minutos.
+* The largest time unit is days, the smallest is minutes.
+* @param {number} futureTime a timestamp in ms to build a phrase for.
+* @param {string} lang Langauge to build the phrase in.
+* @returns {string} A phrase representing the relative time in the future. e.g. 3 days 5 hours.
+*/
+module.exports.formatTimeUntil = (futureTime, lang) => {
```

Comment:
Probably need to rename this since the string will include the "in" part of the phrase as well. Maybe just call it `formatRelativeTime`, unless that's a conflict with something else.

Reply:
Done!